### ISSUE TITLE
libvmaf API does not support `perc*` pool methods

### ISSUE THREAD

**Issue by cosmin** [Netflix/vmaf#818]
The libvmaf API used by ffmpeg does not support the `perc5, perc10, perc20` pooling methods. Furthermore attempting to pass one of these values will silently fallback to mean as the pooling method due to the implementation in `pool_method_map`, leading to a confusing user experience (even though the documentation mentions that only `min`, `mean` and `harmonic_mean` are supported).

It would be great to either unify the pooling methods such that `perc*` works, or to perhaps consider returning an error if an unsupported pooling method is requested.

**Comment by HunterAP23** [comment]
I'm pretty sure the main reason for this is that FFmpeg uses the `compute_vmaf.c` code which is set to be deprecated on the VMAF library. That file only has `min`, `mean` and `harmonic_mean` as per this section:

https://github.com/Netflix/vmaf/blob/bf5e496c4c68875826a9da967f036ddcfc128142/libvmaf/src/compute_vmaf.c#L23-L32

the only other mention of `percX` pooling methods I could find were only present in the python library which uses numpy to get those percentile points. `percX` doesn't seem to be implemented in C/C++ for FFmpeg to import.

**Comment by cosmin** [comment]
Yes, it would require adding percentile support to `vmaf_feature_score_pooled` in `libvmaf.c` but it seems like it would be fairly straightforward.

**Comment by HunterAP23** [comment]
Not trying to plug my code, but I've been working on a python script that takes VMAF report files and turns them into a chart image and a video that graphs the different VMAF values over time. One of the items I provide on the legend is the 1% and 0.1%, and 0.001% lowest percentiles. It's still a work in progress and not yet complete, but if you're willing to use an external program to calculate different percentiles of the different metrics (like PSNR and SSIM) then you can check it out: https://github.com/HunterAP23/VMAF-Over-Time

**Comment by lusoris** [comment]
I submitted the C API portion as #1589, adapted from VMAFx's merged implementation. It adds `VMAF_POOL_METHOD_MEDIAN`, `PERC5`, `PERC10`, and `PERC20`, with linear interpolation matching the Python harness's percentile rule and the existing interval/subsampling selection.

The patch preserves the existing pooling methods and includes five regression cases; the full upstream release suite passes 24/24 and the new cases pass ASan/UBSan/LSan. It is based on current upstream master, not our fork's newer C++/dependency configuration.

FFmpeg still needs its own filter-option mapping change to expose these enum values as `pool=...`; this PR alone does not add those strings to FFmpeg. Also, the silent fallback described in 2021 depended on the caller's mapping: current libvmaf's pooling switch returns `-EINVAL` for an unknown enum rather than silently choosing mean.


Edit 2026-09-19: #1589 has since been corrected. As first submitted it made the XML and JSON report writers read past their pooling-name table; the PR describes the fix, and it now has six regression cases. It was also rebased onto current master.
